### DISCUSSION TITLE
With API create invoice w/ description hash.

### DISCUSSION THREAD

**Discussion by braydonf** [ACINQ/phoenixd#45] [Ideas]
It appears that if you try to create an invoice with a description more than 639 bytes, there will be errors such as "invoice isn't canonically encoded" or "there must be exactly one description tag or one description hash tag". It could be useful here to be able to specify a `descriptionHash` instead of the `description`.

**Comment by braydonf** [comment]
For what it is worth, this is related to adding NIP-57 support w/ phoenixd at https://github.com/braydonf/satdress

**Comment by braydonf** [comment]
It looks like `peer.createInvoice` for the description parameter can either be a string or bytes, if it's bytes (`ByteVector32`) it will be the description hash. So the only changes need are likely in `Api.kt` and perhaps on the responses to include the description hash a field.

**Comment by pm47** [comment]
Released in [v0.1.5](https://github.com/ACINQ/phoenixd/releases/tag/v0.1.5).
